src/components/WelcomeScreen.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+class WelcomeScreen extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {value: ''};
+      this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleSubmit (e) {
+      e.preventDefault();
+      console.log(this.input.value);
+      this.props.setUsername(this.input.value);
```

Comment:
can we find this.input.value on the event obj? I do not think you need a ref here!